lib/puppet/util/network_device/cisco_ios/device.rb

```diff
@@ -36,8 +34,346 @@ def backwards_compatible_schema_load(url_or_config)
       url_or_config
     end
 
+    attr_reader :connection
+    attr_accessor :transport, :facts, :commands
+
+    def send_command(connection_to_use, options, debug = false)
```

Comment:
These changes look like they should be in the transport class https://github.com/puppetlabs/cisco_ios/blob/master/lib/puppet/transport/cisco_ios.rb

And a lot of it looks to be what is available in the transport class already and won't be needed here